flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/catalog/BaseCatalog.java

```diff
@@ -47,35 +67,75 @@ public void close() throws CatalogException {}
 
   @Override
   public List<String> listDatabases() throws CatalogException {
-    throw new UnsupportedOperationException();
+    return Arrays.stream(catalog().asSchemas().listSchemas())
+        .map(NameIdentifier::name)
+        .collect(Collectors.toList());
   }
 
   @Override
-  public CatalogDatabase getDatabase(String s) throws DatabaseNotExistException, CatalogException {
-    throw new UnsupportedOperationException();
+  public CatalogDatabase getDatabase(String databaseName)
+      throws DatabaseNotExistException, CatalogException {
+    try {
+      Schema schema = catalog().asSchemas().loadSchema(databaseName);
+      return new CatalogDatabaseImpl(schema.properties(), schema.comment());
+    } catch (NoSuchSchemaException e) {
+      throw new DatabaseNotExistException(getName(), databaseName);
+    }
   }
 
   @Override
-  public boolean databaseExists(String s) throws CatalogException {
-    throw new UnsupportedOperationException();
+  public boolean databaseExists(String databaseName) throws CatalogException {
+    return catalog().asSchemas().schemaExists(databaseName);
   }
 
   @Override
-  public void createDatabase(String s, CatalogDatabase catalogDatabase, boolean b)
+  public void createDatabase(
+      String databaseName, CatalogDatabase catalogDatabase, boolean ignoreIfExists)
       throws DatabaseAlreadyExistException, CatalogException {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public void dropDatabase(String s, boolean b, boolean b1)
+    try {
+      catalog()
+          .asSchemas()
+          .createSchema(
+              databaseName, catalogDatabase.getComment(), catalogDatabase.getProperties());
+    } catch (SchemaAlreadyExistsException e) {
+      if (!ignoreIfExists) {
+        throw new DatabaseAlreadyExistException(getName(), databaseName);
+      } else {
+        LOG.warn("Database {} already exists.", databaseName);
+      }
+    } catch (NoSuchCatalogException e) {
+      throw new CatalogException(e);
+    }
+  }
+
+  @Override
+  public void dropDatabase(String databaseName, boolean ignoreIfNotExists, boolean cascade)
       throws DatabaseNotExistException, DatabaseNotEmptyException, CatalogException {
-    throw new UnsupportedOperationException();
+    try {
+      catalog().asSchemas().dropSchema(databaseName, cascade);
```

Comment:
if dropSchema returns false, it means no such schema, please handle it explictly.